[REVIEW] Add DGL benchmarks

VibhuJawa opened the pull request:
Results

```
-------------------------------------------------- benchmark: 15 tests ---------------------------------------------------
Name (time in ms, mem in bytes)          Min                 Max                Mean            StdDev            Outliers
--------------------------------------------------------------------------------------------------------------------------
bench_dgl_uva[fanout0-100]            3.5787 (1.0)        5.7552 (1.0)        4.1772 (1.0)      0.6654 (3.51)          2;1
bench_dgl_uva[fanout0-500]            5.9974 (1.68)       7.5749 (1.32)       6.3549 (1.52)     0.5099 (2.69)          1;1
bench_dgl_uva[fanout0-1000]           7.2104 (2.01)       8.2763 (1.44)       7.7201 (1.85)     0.3792 (2.00)          4;0
bench_dgl_uva[fanout0-2500]          13.6467 (3.81)      15.1754 (2.64)      14.0300 (3.36)     0.4769 (2.51)          1;0
bench_dgl_uva[fanout0-5000]          20.2349 (5.65)      21.5314 (3.74)      20.5889 (4.93)     0.4103 (2.16)          1;0
bench_dgl_uva[fanout0-10000]         43.6608 (12.20)     44.8868 (7.80)      43.9273 (10.52)    0.3874 (2.04)          1;1
bench_dgl_uva[fanout0-20000]         65.3379 (18.26)     65.9942 (11.47)     65.5145 (15.68)    0.2191 (1.16)          2;0
bench_dgl_uva[fanout0-30000]         84.8169 (23.70)     85.4202 (14.84)     85.0736 (20.37)    0.1896 (1.0)           3;1
bench_dgl_uva[fanout0-40000]         94.7195 (26.47)     95.2171 (16.54)     94.9354 (22.73)    0.2010 (1.06)          4;0
bench_dgl_uva[fanout0-60000]        120.5435 (33.68)    121.1264 (21.05)    120.7526 (28.91)    0.1947 (1.03)          3;0
bench_dgl_uva[fanout0-50000]        125.1194 (34.96)    126.2888 (21.94)    125.5843 (30.06)    0.3813 (2.01)          3;0
bench_dgl_uva[fanout0-80000]        147.0422 (41.09)    147.6853 (25.66)    147.3432 (35.27)    0.2087 (1.10)          3;0
bench_dgl_uva[fanout0-70000]        158.6202 (44.32)    159.2101 (27.66)    158.8246 (38.02)    0.2099 (1.11)          3;0
bench_dgl_uva[fanout0-100000]       170.6613 (47.69)    171.4075 (29.78)    170.8629 (40.90)    0.2104 (1.11)          1;1
bench_dgl_uva[fanout0-90000]        179.5610 (50.17)    181.3988 (31.52)    180.1198 (43.12)    0.5162 (2.72)          2;1
--------------------------------------------------------------------------------------------------------------------------
```

rlratzel commented:
> Todo: Figure out if UVA can be done via pytest.

Is [this](https://github.com/rapidsai/cugraph/blob/df12979cd25f22b28c52417fc3b3a6a55fcdfc57/benchmarks/cugraph/pytest-based/bench_algos.py#L89-L93) relevant?

VibhuJawa commented:
> Is [this](https://github.com/rapidsai/cugraph/blob/df12979cd25f22b28c52417fc3b3a6a55fcdfc57/benchmarks/cugraph/pytest-based/bench_algos.py#L89-L93) relevant?

@rlratzel , Addressed this, it was an issue with next vs iter which i addressed. Feel free to merge/review again.

rlratzel commented:
@gpucibot merge
